File: eTouristAgencyAPI/eTouristAgencyAPI.Services/Database/Models/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace eTouristAgencyAPI.Services.Database.Models;

[Table("City")]
public partial class City
{
    [Key]
    public Guid Id { get; set; }

    [StringLength(100)]
    public string Name { get; set; } = null!;

    public DateTime CreatedOn { get; set; }

    public DateTime ModifiedOn { get; set; }

    public Guid? CreatedBy { get; set; }

    public Guid? ModifiedBy { get; set; }

    public Guid CountryId { get; set; }

    [ForeignKey("CountryId")]
    [InverseProperty("Cities")]
    public virtual Country Country { get; set; } = null!;

    [ForeignKey("CreatedBy")]
    [InverseProperty("CityCreatedByNavigations")]
    public virtual User? CreatedByNavigation { get; set; }

    [InverseProperty("City")]
    public virtual ICollection<Hotel> Hotels { get; set; } = new List<Hotel>();

    [ForeignKey("ModifiedBy")]
    [InverseProperty("CityModifiedByNavigations")]
    public virtual User? ModifiedByNavigation { get; set; }
}

File: Frontend/etouristagency_mobile/lib/models/city/city.dart
import 'package:etouristagency_mobile/models/country/country.dart';

class City {
  String? id;
  String? name;
  String? countryId;
  Country? country;

  City(this.id, this.name, this.countryId, this.country);

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      json["id"],
      json["name"],
      json["countryId"],
      json["country"] != null ? Country.fromJson(json["country"]) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "countryId": countryId,
      "country": country?.toJson(),
    };
  }
}